platform/android/java/build.gradle

```diff
@@ -122,6 +122,39 @@ task zipCustomBuild(type: Zip) {
     destinationDir(file(binDir))
 }
 
+task generateDevTemplate (type: GradleBuild) {
+    // add parameter to set symbols to true
+    startParameter.projectProperties += [doNotStrip: true]
+
+    // We exclude these gradle tasks so we can run the scons command manually.
+    for (String buildType : supportedTargets) {
+        startParameter.excludedTaskNames += ":lib:" + getSconsTaskName(buildType)
+    }
+
+    tasks = []
+
+    // Only build the apks and aar files for which we have native shared libraries.
+    for (String target : supportedTargets) {
+        File targetLibs = new File("lib/libs/" + target)
+        if (targetLibs != null
+            && targetLibs.isDirectory()
+            && targetLibs.listFiles() != null
+            && targetLibs.listFiles().length > 0) {
+            String capitalizedTarget = target.capitalize()
+            // Copy the generated aar library files to the custom build directory.
+            tasks += "copy" + capitalizedTarget + "AARToAppModule"
+            // Copy the generated aar library files to the bin directory.
+            tasks += "copy" + capitalizedTarget + "AARToBin"
+            // Copy the prebuilt binary templates to the bin directory.
+            tasks += "copy" + capitalizedTarget + "BinaryToBin"
+        } else {
+            logger.lifecycle("No native shared libs for target $target. Skipping build.")
+        }
+    }
+
+    finalizedBy 'zipCustomBuild'
+}
+
```

Comment:
That looks correct, though not ideal due to the duplication of logic between `generateDevTemplate` and `generateGodotTemplate` (e.g: any changes in one location would require a similar change in the other).

Do you know if it would be possible to make `generateDevTemplate` reuse the `generateGodotTemplate` task?
If it's not possible, we could also move the logic within a function and have both tasks just invoke the function.

Reply:
I just couldn't find a way to extend Build gradle task, and every line that uses `startParameters` or `tasks` must be within the task meaning that 99% of the task can't be shared
(At least I couldn't find any way yo do that, I even trued to create a function that get those as parameters but it still didn't work)

Reply:
In that case the approach we can follow is to move the tasks being excluded within its own function. The function will return a list (or array?) of the tasks which we can then assign to `startParameter.excludedTaskNames`.

Then we can also move the logic building the list of tasks to run within its own function. This function here also returns an array of tasks that we can then assign the `tasks` variable.

Reply:
Oh... Sounds like a really good idea 😅 Ill do it in a few hours when Ill be on my pc

Reply:
Ok, fixed